traffic_stats/traffic_stats.go

```diff
@@ -108,6 +112,45 @@ type StartupConfig struct {
 	DailySummaryRetentionPolicy string   `json:"dailySummaryRetentionPolicy"`
 	BpsChan                     chan influx.BatchPoints
 	InfluxDBs                   []*InfluxDBProps
+	KafkaConfig                 KafkaConfig `json:"kafkaConfig"`
+}
+
+type KafkaConfig struct {
+	Enable        bool   `json:"enable"`
+	Brokers       string `json:"brokers"`
+	Topic         string `json:"topic"`
+	RequiredAcks  int    `json:"requiredAcks"`
+	EnableTls     bool   `json:"enableTls"`
+	RootCA        string `json:"rootCA"`
+	ClientCert    string `json:"clientCert"`
+	ClientCertKey string `json:"clientCertKey"`
+}
+
+type KafkaCluster struct {
+	producer *sarama.AsyncProducer
+	client   *sarama.Client
+}
+
+type KafkaJSON struct {
+	Name   string                 `json:"name"`
+	Tags   map[string]string      `json:"tags"`
+	Fields map[string]interface{} `json:"fields"`
+	Time   time.Time              `json:"time"`
+}
+
+type DataExporter interface {
+	ExportData(config StartupConfig, bps influx.BatchPoints, retry bool)
+}
+
+func (c *KafkaCluster) ExportData(config StartupConfig, bps influx.BatchPoints, retry bool) {
+	err := publishToKafka(config, bps, c)
+	if err != nil {
+		log.Errorln("Unable to export to Kafka", err)
+	}
+}
+
+func (influx StartupConfig) ExportData(config StartupConfig, bps influx.BatchPoints, retry bool) {
```

Comment:
Rather than tying this method to `StartupConfig`, it might make more sense to create a new `InfluxClient` struct (or some other name) to associate this interface method with.